gef.py

```diff
@@ -3757,13 +3764,18 @@ def reset_architecture(arch: Optional[str] = None, default: Optional[str] = None
     if not gef.binary:
         gef.binary = get_elf_headers()
 
-    arch_name = gef.binary.e_machine if gef.binary else get_arch()
+    gdb_arch = get_arch()
+    arch_name = gef.binary.e_machine if gef.binary else gdb_arch
 
     if ((arch_name == "MIPS" or arch_name == Elf.Abi.MIPS)
         and (gef.binary is not None and gef.binary.e_class == Elf.Class.ELF_64_BITS)):
         # MIPS64 = arch(MIPS) + 64b flag
         arch_name = "MIPS64"
```

Comment:
Your PR also means the `MIPS64` class can be updated to use `support_gdb_arch()` instead of this ugly hack in the middle of the `get_arch()`.

Reply:
Tested the change with a basic cross-compiled mips64 hello world binary + `qemu-mips64`.